editor/src/components/canvas/canvas-strategies/canvas-strategies.spec.browser2.tsx

```diff
@@ -697,3 +699,222 @@ describe('Snapping guidelines for absolutely moved element', () => {
+          height: 200,
+        }}
+        data-uid='parentsibling'
+        data-testid='parentsibling'
+        data-label='parentsibling'
+      >
+        <div
+          style={{
+            backgroundColor: 'teal',
+            position: 'relative',
+            width: 109,
+            height: 123,
+          }}
+          data-uid='firstdiv'
+          data-testid='firstdiv'
+          data-label='firstdiv'
+        />
+        <div
+          style={{
+            backgroundColor: 'green',
+            position: 'relative',
+            width: 118,
+            height: 123,
+          }}
+          data-uid='thirddiv'
+          data-testid='thirddiv'
+          data-label='thirddiv'
+        />
+      </div>
+    </Scene>
+  </Storyboard>
+)
+`
+
+describe('special case controls', () => {
+  it('non-resizable corner controls show for an element that is not resizable', async () => {
+    const targetElement = elementPath([['storyboard', 'scene', 'sceneroot', 'dragme', 'button']])
+
+    const renderResult = await renderTestEditorWithCode(
+      projectWithNonResizableElement,
+      'await-first-dom-report',
+    )
+
+    await act(async () => {
+      const dispatchDone = renderResult.getDispatchFollowUpActionsFinished()
+      await renderResult.dispatch([selectComponents([targetElement], false)], false)
+      await dispatchDone
+    })
+
+    const nonResizableControl = await renderResult.renderedDOM.findByTestId('non-resizable-control')
+    expect(Prettier.format(nonResizableControl.outerHTML, PrettierConfig)).toEqual(`;<div
+  data-testid='non-resizable-control'
+  style='position: absolute; display: block; left: 197px; top: 376px; width: 100px; height: 30px;'
+>
+  <div style='position: absolute; pointer-events: none;'>
+    <div
+      data-testid='non-resizable-0-0'
+      style='position: relative; width: 6px; height: 6px; top: -3px; left: -3px; background: rgb(255, 255, 255); border: 1px solid rgba(0, 0, 0, 0.5); border-radius: 50%;'
+    ></div>
+  </div>
+  <div style='position: absolute; pointer-events: none; left: 100px;'>
+    <div
+      data-testid='non-resizable-1-0'
+      style='position: relative; width: 6px; height: 6px; top: -3px; left: -3px; background: rgb(255, 255, 255); border: 1px solid rgba(0, 0, 0, 0.5); border-radius: 50%;'
+    ></div>
+  </div>
+  <div style='position: absolute; pointer-events: none; top: 30px;'>
+    <div
+      data-testid='non-resizable-0-1'
+      style='position: relative; width: 6px; height: 6px; top: -3px; left: -3px; background: rgb(255, 255, 255); border: 1px solid rgba(0, 0, 0, 0.5); border-radius: 50%;'
+    ></div>
+  </div>
+  <div style='position: absolute; pointer-events: none; left: 100px; top: 30px;'>
+    <div
+      data-testid='non-resizable-1-1'
+      style='position: relative; width: 6px; height: 6px; top: -3px; left: -3px; background: rgb(255, 255, 255); border: 1px solid rgba(0, 0, 0, 0.5); border-radius: 50%;'
+    ></div>
+  </div>
+</div>
+`)
```

Comment:
could you replace this with one or more more specific assertions? this will break every time we update the design of these indicators, and it's not a snapshot so it'll be a pain to update.

Reply:
Sure.